src/agent/scm/lib/svnwrapper.ts

```diff
@@ -0,0 +1,475 @@
+                        if (shell.test("-d", path.join(subFolder, administrativeDirectoryName))) {
+                            // The subfolder contains .svn directory and we treat it as
+                            // a working copy candidate.
+                            candidates.push(subFolder); 
+                            if (--count == 0) {
+                                deferred.resolve(candidates);
+                            } 
+                        }
+                        else {
+                            // Merge working directory paths found in the subfolder into the common candidates collection.
+                            this._getSvnWorkingCopyPaths(subFolder)
+                            .then((moreCandidates) => {
+                                candidates = candidates.concat(moreCandidates);
+                                if (--count == 0) {
+                                    deferred.resolve(candidates);
+                                } 
+                            })
+                        }
+                        
+                    });
+                }
+                else {
+                    deferred.resolve(candidates);
+                }
+            });
+        }
+        
+        return deferred.promise;
+    }
+
+    private _getTargetUrl(folder: string): Q.Promise<string> {
+        if (!shell.test("-d", folder)) {
+            throw new Error("Folder " + folder + " does not exists");
+        }
+        
+        var deferred = Q.defer<string>();
+        
+        this._shellExec('info', [folder, "--depth", "empty", "--xml"])
+        .then((ret) => {
+            if (!this.isSuccess(ret)) {
+                deferred.resolve(null);
+            }
+            else if (ret.output) {
+                xmlrReader.read(ret.output, (err, res) => {
+                    if (err) {
+                        deferred.reject(err);
+                    }
+                    else {
+                        try {
+                            return deferred.resolve(res.info.entry.url.text());
+                        }
+                        catch (e) {
+                            deferred.reject(e);
+                        }
+                    }
+                });
+            }
+            else {
+                deferred.resolve(null);
+            }
+        });
+        
+        return deferred.promise;
+    }
+
+    public getLatestRevision(sourceBranch: string, sourceRevision: string): Q.Promise<string> {
+        return this._shellExec('info', [this.buildSvnUrl(sourceBranch), 
+                                        "--depth", "empty", 
+                                        "--revision", sourceRevision, 
+                                        "--xml"])
+        .then((ret) => {
+            var defer = Q.defer<any>(); 
+            
+            if (!this.isSuccess(ret)) {
+                defer.reject(ret.output); 
```

Comment:
In this case you never return the promise - that will probably lead to an apparent hang.